rules/no-test-callback.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const getDocsUrl = require('./util').getDocsUrl;
+const isTestCase = require('./util').isTestCase;
+
+module.exports = {
+  meta: {
+    docs: {
+      url: getDocsUrl(__filename),
+    },
+    fixable: 'code',
+  },
+  create(context) {
+    return {
+      CallExpression(node) {
+        if (!isTestCase(node) || node.arguments.length !== 2) {
+          return;
+        }
+
+        const callback = node.arguments[1];
+
+        if (
+          !/^(Arrow)?FunctionExpression$/.test(callback.type) ||
+          callback.params.length !== 1
+        ) {
+          return;
+        }
+
+        const argument = callback.params[0];
+        context.report({
+          node: argument,
+          message: 'Illegal usage of test callback',
```

Comment:
What about something like "Prefer returning a Promise instead of using a test callback"? Not sure if we could be more up front about explaining the issue in the message?

Reply:
Yeah, I actually had that in my first iteration locally, but landed on this to more closely match the name of the rule, and general language of the eslint rules I looked at for the implementation of the fixer (`arrow-parens` and `arrow-body-style`). Thoughts?

>Not sure if we could be more up front about explaining the issue in the message?

How so? Isn't that more the job of the docs?

Reply:
Yeah, I think leaving that detail to the docs is the best option here - makes sense. Thanks for providing that context too. 👍

Reply:
I'd love PRs improving the language of both the lint error and the docs - I suck at both 🙂